$OpenBSD: patch-src_Resolver_cc,v 1.14 2017/02/15 15:27:48 sthen Exp $

"Cast address to IN6_IS_ADDR_LINKLOCAL."

backport from github 259d642e1fea2ddf38763d49e8e7701f0a947d4c

--- src/Resolver.cc.orig	Mon Jan 16 21:44:52 2017
+++ src/Resolver.cc	Mon Jan 16 21:45:30 2017
@@ -318,7 +318,7 @@ void Resolver::AddAddress(int family,const char *addre
    case AF_INET6:
       if(sizeof(add.in6.sin6_addr) != len)
          return;
-      if(IN6_IS_ADDR_LINKLOCAL(address) && scope==0) {
+      if(IN6_IS_ADDR_LINKLOCAL((const struct in6_addr*)address) && scope==0) {
 	 error=_("Link-local IPv6 address should have a scope");
 	 return;
       }
